homeassistant/components/vizio/config_flow.py

```diff
@@ -101,6 +114,25 @@ def __init__(self) -> None:
         """Initialize config flow."""
         self._user_schema = None
         self._must_show_form = None
+        self._ch_type = None
+        self._pairing_token = None
+        self._data = None
+
+    async def _create_entry_if_unique(
+        self, input_dict: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Check if unique_id doesn't already exist. If it does, abort. If it doesn't, create entry."""
+        unique_id = await VizioAsync.get_unique_id(
+            input_dict[CONF_HOST],
+            input_dict.get(CONF_ACCESS_TOKEN),
+            input_dict[CONF_DEVICE_CLASS],
+            session=async_get_clientsession(self.hass, False),
+        )
+
+        if await self.async_set_unique_id(unique_id=unique_id, raise_on_progress=True):
+            return self.async_abort(reason="already_setup_with_diff_host_and_name")
```

Comment:
This is not necessary, `async_set_unique_id` will already raise.

Reply:
This if statement checks to ensure an existing entry with the same `unique_id` doesn't already exist. I believe `async_set_unique_id` only raises an Exception when an in progress config flow exists with the same `unique_id`

Reply:
You can use self._abort_if_unique_id_configured